src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeoAuthProvider.scala

```diff
@@ -0,0 +1,39 @@
+package org.broadinstitute.dsde.workbench.leonardo.model
+
+import java.util.UUID
+
+import com.typesafe.config.Config
+import org.broadinstitute.dsde.workbench.leonardo.api.AuthorizationError
+import org.broadinstitute.dsde.workbench.model.WorkbenchUserEmail
+
+import scala.concurrent.Future
+
+sealed trait ProjectAction
+case object ListClusters extends ProjectAction
+case object CreateClusters extends ProjectAction
+
+sealed trait NotebookClusterAction
+case object GetClusterDetails extends NotebookClusterAction
+case object ConnectToCluster extends NotebookClusterAction
+case object LocalizeDataToCluster extends NotebookClusterAction
+case object DestroyCluster extends NotebookClusterAction
+
+abstract class LeoAuthProvider(authConfig: Config) {
+  //Does this user have permission in all projects to perform this action?
```

Comment:
Also kind of minor, but since these interfaces are externally facing, should we provide actual Scaladoc for them?

Reply:
Yes. I thought this as I wrote it but wanted to get my ideas down quickly. Will tidy this before the PR is ready for reals.